### PR TITLE
feat: [CDS-73656]: Add poller for trigger events from queue service

### PR DESCRIPTION
**Pull request by shalini-agr**
## Checklist
- [ ] I've documented the changes in the PR description.
- [ ] I've tested this change either in PR or local environment.
- [ ] If hashcheck failed I followed [the checklist](https://harness.atlassian.net/wiki/spaces/DEL/pages/21016838831/PR+Codebasehash+Check+merge+checklist) and updated this PR description with my answers to make sure I'm not introducing any breaking changes.

## Comment Triggers
<details>
  <summary>Build triggers</summary>

- Feature build: `trigger feature-build`
- 
  Specific builds
  - `trigger manager`
  - `trigger dms`
  - `trigger ng_manager`
  - `trigger cvng `
  - `trigger cmdlib`
  - `trigger template_svc`
  - `trigger events_fmwrk_monitor`
  - `trigger event_server`
  - `trigger change_data_capture`
  -  Trigger multiple builds together: For eg: `trigger dms, manager`
- Immutable delegate `trigger publish-delegate`
</details>

## [Latest PR Check Triggers](https://github.com/harness/harness-core/blob/develop/.github/pull_request_template.md)

<details>
  <summary>PR Check triggers</summary>
You can run multiple PR check triggers by comma separating them in a single comment. e.g. `trigger ut0, ut1`

- Compile: `trigger compile`
- CodeformatCheckstyle: `trigger checkstylecodeformat`
  - CodeFormat: `trigger codeformat`
  - Checkstyle: `trigger checkstyle`
- MessageMetadata: `trigger messagecheck`
- File-Permission-Check: `trigger checkpermission`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- Trigger CommonChecks: `trigger commonchecks`
- PMD: `trigger pmd`
- Copyright Check: `trigger copyrightcheck`
- Feature Name Check: `trigger featurenamecheck`
- UnitTests-ALL: `trigger uts`
- UnitTests-0: `trigger ut0`
- UnitTests-1: `trigger ut1`
- UnitTests-2: `trigger ut2`
- UnitTests-3: `trigger ut3`
- UnitTests-4: `trigger ut4`
- UnitTests-5: `trigger ut5`
- UnitTests-6: `trigger ut6`
- UnitTests-7: `trigger ut7`
- UnitTests-8: `trigger ut8`
- UnitTests-9: `trigger ut9`
- CodeBaseHash: `trigger codebasehash`
- CodeFormatCheckstyle: `trigger checkstylecodeformat`
- SonarScan: `trigger ss`
- GitLeaks: `trigger gitleaks`
- Trigger all Checks: `trigger smartchecks`
- Go Build: `trigger gobuild`
- Validate_Reviews: `trigger review`
- Module Dependency Check: `trigger mdc`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

## [Contributor license agreement](https://github.com/harness/harness-core/blob/develop/CONTRIBUTOR_LICENSE_AGREEMENT.md)

- - -
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/50353)

### CODE REVIEW

**Comment by shalini-agr** [comment]
trigger uts

**Comment by shalini-agr** [comment]
trigger messagecheck

File: 878-ng-common-utilities/src/main/java/io/harness/queuePoller/AbstractEventQueueProcessor.java [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+                                                                     .batchSize(batchSize)
+                                                                     .consumerName(getTopicName())
+                                                                     .topic(getTopicName())
+                                                                     .maxWaitDuration(100)
+                                                                     .build());
+      for (DequeueResponse message : messages) {
+        processMessage(message);
+      }
+    } catch (Exception e) {
+      throw new RuntimeException(e);
```
Comment (ankittiwari-harness) [inline 878-ng-common-utilities/src/main/java/io/harness/queuePoller/AbstractEventQueueProcessor.java:90]: nit: print exception

File: pipeline-service/service/src/main/java/io/harness/pms/event/webhookevent/WebhookEventQueueProcessor.java [hunk 2]
```diff
@@ -0,0 +1,176 @@
...
+import java.util.Objects;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.atomic.AtomicBoolean;
+import lombok.extern.slf4j.Slf4j;
+
+@Slf4j
+@OwnedBy(HarnessTeam.PIPELINE)
+public class WebhookEventQueueProcessor implements Managed {
```
Comment (ankittiwari-harness) [inline pipeline-service/service/src/main/java/io/harness/pms/event/webhookevent/WebhookEventQueueProcessor.java:52]: why can't we extend AbstractEventQueueProcessor here

**Review by shalini-agr — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/50353)* status: 9 file(s) to review. 1. unresolved discussions. 0 of 2 approvals obtained. 

___
*[`pipeline-service/service/src/main/java/io/harness/pms/event/webhookevent/WebhookEventQueueProcessor.java` line 52 at r2](https://reviewable.io/reviews/harness/harness-core/50353#gh-1263332103:-N_I9wPc5YZSwmZPkZ-u:b-auzmvw) ([raw file](https://github.com/harness/harness-core/blob/689106297828d0a14387c1c09ca249e4fe0ab885/pipeline-service/service/src/main/java/io/harness/pms/event/webhookevent/WebhookEventQueueProcessor.java#L52)):*
<details><summary><i>Previously, ankittiwari-harness (Ankit Tiwari) wrote…</i></summary><blockquote>

why can't we extend AbstractEventQueueProcessor here
</blockquote></details>

yes, we can, 
done

**Review by archit-harness — CHANGES_REQUESTED** [review]
> Reviewed 2 of 7 files at r2, 8 of 10 files at r3, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/50353)* status: 14 file(s) to review. 1 change requests. 1. unresolved discussions. 0 of 2 approvals obtained. 

___
*[`878-ng-common-utilities/src/main/java/io/harness/queuePoller/AbstractEventQueueProcessor.java` line 37 at r3](https://reviewable.io/reviews/harness/harness-core/50353#-N_ITuJP8_WYqzlZOy02:-N_ITuJP8_WYqzlZOy03:b-k6xo97) ([raw file](https://github.com/harness/harness-core/blob/e000c7c194f0586ee16814ac4df089c7627e62ca/878-ng-common-utilities/src/main/java/io/harness/queuePoller/AbstractEventQueueProcessor.java#L37)):*
> ```Java
> @Slf4j
> @OwnedBy(HarnessTeam.PIPELINE)
> public abstract class AbstractEventQueueProcessor implements Managed {
> ```

AbstractHsqsQueueProcessor
___
*[`878-ng-common-utilities/src/main/java/io/harness/queuePoller/AbstractEventQueueProcessor.java` line 42 at r3](https://reviewable.io/reviews/harness/harness-core/50353#-N_IW-pt5F4HX6Yg3tVb:-N_IW-pt5F4HX6Yg3tVc:b-9di7w9) ([raw file](https://github.com/harness/harness-core/blob/e000c7c194f0586ee16814ac4df089c7627e62ca/878-ng-common-utilities/src/main/java/io/harness/queuePoller/AbstractEventQueueProcessor.java#L42)):*
> ```Java
> 
>   private static final int WAIT_TIME_IN_SECONDS = 5;
>   private final int batchSize = 5;
> ```

We should increase to

**Review by shalini-agr — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/50353)* status: 19 file(s) to review. 1 change requests. 3. unresolved discussions. 0 of 2 approvals obtained. 

___
*[`878-ng-common-utilities/src/main/java/io/harness/queuePoller/AbstractEventQueueProcessor.java` line 37 at r3](https://reviewable.io/reviews/harness/harness-core/50353#-N_ITuJP8_WYqzlZOy02:-N_JLenKDAKSM82UNfqw:b-896fix) ([raw file](https://github.com/harness/harness-core/blob/e000c7c194f0586ee16814ac4df089c7627e62ca/878-ng-common-utilities/src/main/java/io/harness/queuePoller/AbstractEventQueueProcessor.java#L37)):*
<details><summary><i>Previously, archit-harness (Archit Singla) wrote…</i></summary><blockquote>

AbstractHsqsQueueProcessor
</blockquote></details>

Done.
___
*[`878-ng-common-utilities/src/main/java/io/harness/queuePoller/AbstractEventQueueProcessor.java` line 42 at r3](https://reviewable.io/reviews/harness/harness-core/50353#-N_IW-pt5F4HX6Yg3tVb:-N_JY-LP5XZvObXjkL_g:b-896fix) ([raw file](https://github.com/harness/harness-core/blob/e000c7c194f0586ee16814ac4df089c7627e62ca/878-ng-common-utilities/src/main/java/io/harness/queuePoller/AbstractEventQueueProcessor.java#L42)):*
<details><summary><i>Previously, archit-harness (Archit Singla) wrote…</i></summary><blockquote>

Discussed offline to take configs from respective pollers
</blockquote></details>

Done.

**Comment by shalini-agr** [comment]
trigger uts

**Review by archit-harness — CHANGES_REQUESTED** [review]
> Reviewed 1 of 7 files at r1, 3 of 12 files at r4, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/50353)* status: 19 file(s) to review. 1 change requests. 1. unresolved discussions. 0 of 2 approvals obtained. 

___
*[`120-ng-manager/config.yml` line 546 at r4](https://reviewable.io/reviews/harness/harness-core/50353#-N_XPVmYDK7nWlvPmj02:-N_XPVmYDK7nWlvPmj03:bfoze4) ([raw file](https://github.com/harness/harness-core/blob/1aa3b42221d3cadcf67da3373cc37be0aba5cf94/120-ng-manager/config.yml#L546)):*
> ```YAML
> webhookPushEventHsqsDequeueConfig:
>   batchSize: {WEBHOOK_PUSH_EVENT_DEQUE_BATCH_SIZE:-20}
>   THREAD_SLEEP_TIME_IN_MILLIS: {WEBHOOK_PUSH_EVENT_DEQUE_THREAD_SLEEP_TIME_IN_MILLIS:-300}
> ```

it should follow camel case
___
*[`120-ng-manager/config.yml` line 550 at r4](https://reviewable.io/reviews/harness/harness-core/50353#-N_XPYhb8A1re7e7-EyT:-N_XPYhb8A1re7e7-EyU:bfoze4) ([raw file](https://github.com/harness/harness-core/blob/1aa3b42221d3cadcf67da3373cc37be0aba5cf94/120-ng-manager/config.yml#L550)):*
> ```YAML
> webhookBranchHookEventHsqsDequeueConfig:
>   batchSize: {WEBHOOK_BRANCH_HOOK_EVENT_DEQUE_BATCH_SIZE:-20}
>   THREAD_SLEEP_TIME_IN_MILLIS: {WEBHOOK_BRANCH_HOOK_EVENT_DEQUE_THREAD_SLEEP_TIME_IN_MILLIS:-300}
> ```

it should follow camel case
___
*[`clients/queue-service/java-client/src/main/java/io/harness/hsqs/client/beans/HsqsDequeueConfig.java` line 22 at r4](https://reviewable.io/reviews/harness/harness-core/50353#-N_XSuLQE_nwB_q0Xrh7:-N_XSuLQE_nwB_q0Xrh8:bpezyy8) ([raw file](https://github.com/harness/harness-core/blob/1aa3b42221d3cadcf67da3373cc37be0aba5cf94/clients/queue-service/java-client/src/main/java/io/harness/hsqs/client/beans/HsqsDequeueConfig.java#L22)):*
> ```Java
> public class HsqsDequeueConfig {
>   @JsonProperty(defaultValue = "20") @Builder.Default int batchSize = 20;
>   @JsonProperty(defaultValue = "300") @Builder.Default int THREAD_SLEEP_TIME_IN_MILLIS = 300;
> ```

follow camel case only as its only static constant
___
*[`pipeline-service/config/config.yml` line 658 at r4](https://reviewable.io/reviews/harness/harness-core/50353#-N_XSxwtAtZo-ncTPqd7:-N_XSxwtAtZo-ncTPqd8:badnndr) ([raw file](https://github.com/harness/harness-core/blob/1aa3b42221d3cadcf67da3373cc37be0aba5cf94/pipeline-service/config/config.yml#L658)):*
> ```YAML
> webhookEventHsqsDequeueConfig:
>   batchSize: {WEBHOOK_EVENT_DEQUE_BATCH_SIZE:-20}
>   THREAD_SLEEP_TIME_IN_MILLIS: {WEBHOOK_EVENT_DEQUE_THREAD_SLEEP_TIME_IN_MILLIS:-300}
> ```

camel case
___
*[`pipeline-service/service/src/main/java/io/harness/pms/event/webhookevent/WebhookEventQueueProcessor.java` line 41 at r4](https://reviewable.io/reviews/harness/harness-core/50353#-N_XT-dG4cyK8qk5_UUF:-N_XT-dG4cyK8qk5_UUG:bpf92k8) ([raw file](https://github.com/harness/harness-core/blob/1aa3b42221d3cadcf67da3373cc37be0aba5cf94/pipeline-service/service/src/main/java/io/harness/pms/event/webhookevent/WebhookEventQueueProcessor.java#L41)):*
> ```Java
> public class WebhookEventQueueProcessor extends AbstractHsqsQueueProcessor {
>   @Inject TriggerWebhookExecutionServiceV2 triggerWebhookExecutionServiceV2;
>   @Inject PipelineServiceConfiguration pipelineServiceConfiguration;
> ```

same as ng one
___
*[`120-ng-manager/src/main/java/io/harness/ng/webhook/queuePoller/WebhookBranchHookEventQueueProcessor.java` line 8 at r4](https://reviewable.io/reviews/harness/harness-core/50353#-N_XRXEb4X0oo-wQzmSn:-N_XRXEb4X0oo-wQzmSo:bu9utx2) ([raw file](https://github.com/harness/harness-core/blob/1aa3b42221d3cadcf67da3373cc37be0aba5cf94/120-ng-manager/src/main/java/io/harness/ng/webhook/queuePoller/WebhookBranchHookEventQueueProcessor.java#L8)):*
> ```Java
>  */
> 
> package io.harness.ng.webhook.queuePoller;
> ```

no need to move packages then
___
*[`120-ng-manager/src/main/java/io/harness/ng/webhook/queuePoller/WebhookBranchHookEventQueueProcessor.java` line 33 at r4](https://reviewable.io/reviews/harness/harness-core/50353#-N_XPfEJDnKegbqB_OVg:-N_XPfEJDnKegbqB_OVh:b-3lbqdq) ([raw file](https://github.com/harness/harness-core/blob/1aa3b42221d3cadcf67da3373cc37be0aba5cf94/120-ng-manager/src/main/java/io/harness/ng/webhook/queuePoller/WebhookBranchHookEventQueueProcessor.java#L33)):*
> ```Java
> public class WebhookBranchHookEventQueueProcessor extends AbstractHsqsQueueProcessor {
>   @Inject GitBranchHookEventExecutionService gitBranchHookEventExecutionService;
>   @Inject NextGenConfiguration nextGenConfiguration;
> ```

we shouldnt pass nextgen configuration to the processor?

Expose these config in NextGenModule
___
*[`120-ng-manager/src/main/java/io/harness/ng/webhook/queuePoller/WebhookPushEventQueueProcessor.java` line 9 at r4](https://reviewable.io/reviews/harness/harness-core/50353#-N_XRZkQ7MD6lqlnD05I:-N_XRZkQ7MD6lqlnD05J:bu9utx2) ([raw file](https://github.com/harness/harness-core/blob/1aa3b42221d3cadcf67da3373cc37be0aba5cf94/120-ng-manager/src/main/java/io/harness/ng/webhook/queuePoller/WebhookPushEventQueueProcessor.java#L9)):*
> ```Java
> 
> package io.harness.ng.webhook.queuePoller;
> ```

no need to move packages then
___
*[`120-ng-manager/src/test/java/io/harness/ng/webhook/queuePoller/WebhookBranchHookEventQueueProcessorTest.java` line 8 at r4](https://reviewable.io/reviews/harness/harness-core/50353#-N_XR_TjE20d0IeNYTzC:-N_XR_TjE20d0IeNYTzD:bu9utx2) ([raw file](https://github.com/harness/harness-core/blob/1aa3b42221d3cadcf67da3373cc37be0aba5cf94/120-ng-manager/src/test/java/io/harness/ng/webhook/queuePoller/WebhookBranchHookEventQueueProcessorTest.java#L8)):*
> ```Java
>  */
> 
> package io.harness.ng.webhook.queuePoller;
> ```

no need to move packages then
___
*[`120-ng-manager/src/test/java/io/harness/ng/webhook/queuePoller/WebhookPushEventQueueProcessorTest.java` line 7 at r4](https://reviewable.io/reviews/harness/harness-core/50353#-N_XR_rNBG0koRGur8ey:-N_XR_rNBG0koRGur8ez:bu9utx2) ([raw file](https://github.com/harness/harness-core/blob/1aa3b42221d3cadcf67da3373cc37be0aba5cf94/120-ng-manager/src/test/java/io/harness/ng/webhook/queuePoller/WebhookPushEventQueueProcessorTest.java#L7)):*
> ```Java
>  * https://polyformproject.org/wp-content/uploads/2020/05/PolyForm-Free-Trial-1.0.0.txt.
>  */
> ```

no need to move packages then
___
*[`878-ng-common-utilities/src/main/java/io/harness/queuePoller/AbstractHsqsQueueProcessor.java` line 42 at r4](https://reviewable.io/reviews/harness/harness-core/50353#-N_XRcJ71d-HIB6ofpoK:-N_XRcJ71d-HIB6ofpoL:buk9q5h) ([raw file](https://github.com/harness/harness-core/blob/1aa3b42221d3cadcf67da3373cc37be0aba5cf94/878-ng-common-utilities/src/main/java/io/harness/queuePoller/AbstractHsqsQueueProcessor.java#L42)):*
> ```Java
>   @Inject HsqsClientService hsqsClientService;
> 
>   private static final int WAIT_TIME_IN_SECONDS = 1;
> ```

rename to reflect this is queue_fetch_wait_time
___
*[`878-ng-common-utilities/src/main/java/io/harness/queuePoller/AbstractHsqsQueueProcessor.java` line 43 at r4](https://reviewable.io/reviews/harness/harness-core/50353#-N_XRho-7r27MjGjxOzS:-N_XRho-7r27MjGjxOzT:b-tqi5dl) ([raw file](https://github.com/harness/harness-core/blob/1aa3b42221d3cadcf67da3373cc37be0aba5cf94/878-ng-common-utilities/src/main/java/io/harness/queuePoller/AbstractHsqsQueueProcessor.java#L43)):*
> ```Java
> 
>   private static final int WAIT_TIME_IN_SECONDS = 1;
>   private static final int SLEEP_SECONDS = 10;
> ```

rename this to reflect this is MAINTENANCE FLAG SLEEP
___
*[`878-ng-common-utilities/src/main/java/io/harness/queuePoller/AbstractHsqsQueueProcessor.java` line 90 at r4](https://reviewable.io/reviews/harness/harness-core/50353#-N_XSPejA8htxLrZo9X_:-N_XSPejA8htxLrZo9Xa:b-gbgwtt) ([raw file](https://github.com/harness/harness-core/blob/1aa3b42221d3cadcf67da3373cc37be0aba5cf94/878-ng-common-utilities/src/main/java/io/harness/queuePoller/AbstractHsqsQueueProcessor.java#L90)):*
> ```Java
>       }
>       if (messages.size() < getHsqsDequeueConfig().getBatchSize()) {
>         TimeUnit.MILLISECONDS.sleep(getHsqsDequeueConfig().getTHREAD_SLEEP_TIME_IN_MILLIS());
> ```

if message size equal to batch size, then sleep for 30ms, this config should be present in config and 300ms one can be constant in this consumer for now, extract this as method in this class to have clean code

**Review by shalini-agr — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/50353)* status: 21 file(s) to review. 1 change requests. 14. unresolved discussions. 0 of 2 approvals obtained. 

___
*[`120-ng-manager/config.yml` line 546 at r4](https://reviewable.io/reviews/harness/harness-core/50353#-N_XPVmYDK7nWlvPmj02:-N_XYZiT01FsCHN997qr:b-896fix) ([raw file](https://github.com/harness/harness-core/blob/1aa3b42221d3cadcf67da3373cc37be0aba5cf94/120-ng-manager/config.yml#L546)):*
<details><summary><i>Previously, archit-harness (Archit Singla) wrote…</i></summary><blockquote>

it should follow camel case
</blockquote></details>

Done.
___
*[`120-ng-manager/config.yml` line 550 at r4](https://reviewable.io/reviews/harness/harness-core/50353#-N_XPYhb8A1re7e7-EyT:-N_XYZw0BtS5VlHM5XNd:b-896fix) ([raw file](https://github.com/harness/harness-core/blob/1aa3b42221d3cadcf67da3373cc37be0aba5cf94/120-ng-manager/config.yml#L550)):*
<details><summary><i>Previously, archit-harness (Archit Singla) wrote…</i></summary><blockquote>

it should follow camel case
</blockquote></details>

Done.
___
*[`clients/queue-service/java-client/src/main/java/io/harness/hsqs/client/beans/HsqsDequeueConfig.java` line 22 at r4](https://reviewable.io/reviews/harness/harness-core/50353#-N_XSuLQE_nwB_q0Xrh7:-N_XcaF62W6iaFwzUdDx:b-896fix) ([raw file](https://github.com/harness/harness-core/blob/1aa3b42221d3cadcf67da3373cc37be0aba5cf94/clients/queue-service/java-client/src/main/java/io/harness/hsqs/client/beans/HsqsDequeueConfig.java#L22)):*
<details><summary><i>Previously, archit-harness (Archit Singla) wrote…</i></summary><blockquote>

follow camel case only as its only static constant
</blockquote></details>

Done.
___
*[`pipeline-service/config/config.yml` line 658 at r4](https://reviewable.io/reviews/harness/harness-core/50353#-N_XSxwtAtZo-ncTPqd7:-N_Xcc1G-Lu_wZDxSGTa:b-896fix) ([raw file](https://github.com/harness/harness-core/blob/1aa3b42221d3cadcf67da3373cc37be0aba5cf94/pipeline-service/config/config.yml#L658)):*
<details><summary><i>Previously, archit-harness (Archit Singla) wrote…</i></summary><blockquote>

camel case
</blockquote></details>

Done.
___
*[`pipeline-service/service/src/main/java/io/harness/pms/event/webhookevent/WebhookEventQueueProcessor.java` line 41 at r4](https://reviewable.io/reviews/harness/harness-core/50353#-N_XT-dG4cyK8qk5_UUF:-N_XlnntAs0ZaJXnntNP:b-896fix) ([raw file](https://github.com/harness/harness-core/blob/1aa3b42221d3cadcf67da3373cc37be0aba5cf94/pipeline-service/service/src/main/java/io/harness/pms/event/webhookevent/WebhookEventQueueProcessor.java#L41)):*
<details><summary><i>Previously, archit-harness (Archit Singla) wrote…</i></summary><blockquote>

same as ng one
</blockquote></details>

Done.
___
*[`120-ng-manager/src/main/java/io/harness/ng/webhook/queuePoller/WebhookBranchHookEventQueueProcessor.java` line 8 at r4](https://reviewable.io/reviews/harness/harness-core/50353#-N_XRXEb4X0oo-wQzmSn:-N_XloWjDXC-1sYQi4KK:b-896fix) ([raw file](https://github.com/harness/harness-core/blob/1aa3b42221d3cadcf67da3373cc37be0aba5cf94/120-ng-manager/src/main/java/io/harness/ng/webhook/queuePoller/WebhookBranchHookEventQueueProcessor.java#L8)):*
<details><summary><i>Previously, archit-harness (Archit Singla) wrote…</i></summary><blockquote>

no need to move packages then
</blockquote></details>

Done.
___
*[`120-ng-manager/src/main/java/io/harness/ng/webhook/queuePoller/WebhookBranchHookEventQueueProcessor.java` line 33 at r4](https://reviewable.io/reviews/harness/harness-core/50353#-N_XPfEJDnKegbqB_OVg:-N_XloyT-UIHRpp5az7b:b-896fix) ([raw file](https://github.com/harness/harness-core/blob/1aa3b42221d3cadcf67da3373cc37be0aba5cf94/120-ng-manager/src/main/java/io/harness/ng/webhook/queuePoller/WebhookBranchHookEventQueueProcessor.java#L33)):*
<details><summary><i>Previously, archit-harness (Archit Singla) wrote…</i></summary><blockquote>

we shouldnt pass nextgen configuration to the processor?

Expose these config in NextGenModule
</blockquote></details>

Done.
___
*[`120-ng-manager/src/main/java/io/harness/ng/webhook/queuePoller/WebhookPushEventQueueProcessor.java` line 9 at r4](https://reviewable.io/reviews/harness/harness-core/50353#-N_XRZkQ7MD6lqlnD05I:-N_XlpgO6if8Dmc1A8Oh:b-896fix) ([raw file](https://github.com/harness/harness-core/blob/1aa3b42221d3cadcf67da3373cc37be0aba5cf94/120-ng-manager/src/main/java/io/harness/ng/webhook/queuePoller/WebhookPushEventQueueProcessor.java#L9)):*
<details><summary><i>Previously, archit-harness (Archit Singla) wrote…</i></summary><blockquote>

no need to move packages then
</blockquote></details>

Done.
___
*[`120-ng-manager/src/test/java/io/harness/ng/webhook/queuePoller/WebhookBranchHookEventQueueProcessorTest.java` line 8 at r4](https://reviewable.io/reviews/harness/harness-core/50353#-N_XR_TjE20d0IeNYTzC:-N_XllkV0IPFXKEiOoAz:b-896fix) ([raw file](https://github.com/harness/harness-core/blob/1aa3b42221d3cadcf67da3373cc37be0aba5cf94/120-ng-manager/src/test/java/io/harness/ng/webhook/queuePoller/WebhookBranchHookEventQueueProcessorTest.java#L8)):*
<details><summary><i>Previously, archit-harness (Archit Singla) wrote…</i></summary><blockquote>

no need to move packages then
</blockquote></details>

Done.
___
*[`120-ng-manager/src/test/java/io/harness/ng/webhook/queuePoller/WebhookPushEventQueueProcessorTest.java` line 7 at r4](https://reviewable.io/reviews/harness/harness-core/50353#-N_XR_rNBG0koRGur8ey:-N_Xlm8RAxCoxoZIyytt:b-896fix) ([raw file](https://github.com/harness/harness-core/blob/1aa3b42221d3cadcf67da3373cc37be0aba5cf94/120-ng-manager/src/test/java/io/harness/ng/webhook/queuePoller/WebhookPushEventQueueProcessorTest.java#L7)):*
<details><summary><i>Previously, archit-harness (Archit Singla) wrote…</i></summary><blockquote>

no need to move packages then
</blockquote></details>

Done.
___
*[`878-ng-common-utilities/src/main/java/io/harness/queuePoller/AbstractHsqsQueueProcessor.java` line 42 at r4](https://reviewable.io/reviews/harness/harness-core/50353#-N_XRcJ71d-HIB6ofpoK:-N_XYvCF8WqpCPSe99yD:b-896fix) ([raw file](https://github.com/harness/harness-core/blob/1aa3b42221d3cadcf67da3373cc37be0aba5cf94/878-ng-common-utilities/src/main/java/io/harness/queuePoller/AbstractHsqsQueueProcessor.java#L42)):*
<details><summary><i>Previously, archit-harness (Archit Singla) wrote…</i></summary><blockquote>

rename to reflect this is queue_fetch_wait_time
</blockquote></details>

Done.
___
*[`878-ng-common-utilities/src/main/java/io/harness/queuePoller/AbstractHsqsQueueProcessor.java` line 43 at r4](https://reviewable.io/reviews/harness/harness-core/50353#-N_XRho-7r27MjGjxOzS:-N_XYvOP4hQbOC8_jsU1:b-896fix) ([raw file](https://github.com/harness/harness-core/blob/1aa3b42221d3cadcf67da3373cc37be0aba5cf94/878-ng-common-utilities/src/main/java/io/harness/queuePoller/AbstractHsqsQueueProcessor.java#L43)):*
<details><summary><i>Previously, archit-harness (Archit Singla) wrote…</i></summary><blockquote>

rename this to reflect this is MAINTENANCE FLAG SLEEP
</blockquote></details>

Done.
___
*[`878-ng-common-utilities/src/main/java/io/harness/queuePoller/AbstractHsqsQueueProcessor.java` line 90 at r4](https://reviewable.io/reviews/harness/harness-core/50353#-N_XSPejA8htxLrZo9X_:-N_Xlmx4D9zRBsSUqhx_:b-896fix) ([raw file](https://github.com/harness/harness-core/blob/1aa3b42221d3cadcf67da3373cc37be0aba5cf94/878-ng-common-utilities/src/main/java/io/harness/queuePoller/AbstractHsqsQueueProcessor.java#L90)):*
<details><summary><i>Previously, archit-harness (Archit Singla) wrote…</i></summary><blockquote>

if message size equal to batch size, then sleep for 30ms, this config should be present in config and 300ms one can be constant in this consumer for now, extract this as method in this class to have clean code
</blockquote></details>

Done.

**Review by archit-harness — CHANGES_REQUESTED** [review]
> Reviewed 11 of 12 files at r5, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/50353)* status: 21 file(s) to review. 1 change requests. 1. unresolved discussions. 0 of 2 approvals obtained. 

___
*[`clients/queue-service/java-client/src/main/java/io/harness/hsqs/client/beans/HsqsDequeueConfig.java` line 22 at r5](https://reviewable.io/reviews/harness/harness-core/50353#-N_Y07BQ1_mtGGYjbWnN:-N_Y07BQ1_mtGGYjbWnO:b87mmbl) ([raw file](https://github.com/harness/harness-core/blob/5b042227d920a9f80923cc0ae53d7454f2f30006/clients/queue-service/java-client/src/main/java/io/harness/hsqs/client/beans/HsqsDequeueConfig.java#L22)):*
> ```Java
> public class HsqsDequeueConfig {
>   @JsonProperty(defaultValue = "20") @Builder.Default int batchSize = 20;
>   @JsonProperty(defaultValue = "30") @Builder.Default int threadSleepTimeInMillis = 30;
> ```

add a comment, that this sleep is if message size is equal to batch size
___
*[`878-ng-common-utilities/src/main/java/io/harness/queuePoller/AbstractHsqsQueueProcessor.java` line 130 at r5](https://reviewable.io/reviews/harness/harness-core/50353#-N_Y01E3E5GrB4Ja0KD1:-N_Y01E3E5GrB4Ja0KD2:b-f9eb8t) ([raw file](https://github.com/harness/harness-core/blob/5b042227d920a9f80923cc0ae53d7454f2f30006/878-ng-common-utilities/src/main/java/io/harness/queuePoller/AbstractHsqsQueueProcessor.java#L130)):*
> ```Java
>       } else {
>         TimeUnit.MILLISECONDS.sleep(getHsqsDequeueConfig().getThreadSleepTimeInMillis());
>       }
> ```

can you check what if this value is made to 0, will it handle? if not handle it



_Code quote:_
```Java
      } else {
        TimeUnit.MILLISECONDS.sleep(getHsqsDequeueConfig().getThreadSleepTimeInMillis());
      }
```

**Review by shalini-agr — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/50353)* status: 21 file(s) to review. 1 change requests. 3. unresolved discussions. 0 of 2 approvals obtained. 

___
*[`clients/queue-service/java-client/src/main/java/io/harness/hsqs/client/beans/HsqsDequeueConfig.java` line 22 at r5](https://reviewable.io/reviews/harness/harness-core/50353#-N_Y07BQ1_mtGGYjbWnN:-N_Y5BEbEaaUuma6CC-N:b-896fix) ([raw file](https://github.com/harness/harness-core/blob/5b042227d920a9f80923cc0ae53d7454f2f30006/clients/queue-service/java-client/src/main/java/io/harness/hsqs/client/beans/HsqsDequeueConfig.java#L22)):*
<details><summary><i>Previously, archit-harness (Archit Singla) wrote…</i></summary><blockquote>

add a comment, that this sleep is if message size is equal to batch size
</blockquote></details>

Done.
___
*[`878-ng-common-utilities/src/main/java/io/harness/queuePoller/AbstractHsqsQueueProcessor.java` line 130 at r5](https://reviewable.io/reviews/harness/harness-core/50353#-N_Y01E3E5GrB4Ja0KD1:-N_Y6qFSBccKq1dWpcLg:b-6q61od) ([raw file](https://github.com/harness/harness-core/blob/5b042227d920a9f80923cc0ae53d7454f2f30006/878-ng-common-utilities/src/main/java/io/harness/queuePoller/AbstractHsqsQueueProcessor.java#L130)):*
<details><summary><i>Previously, archit-harness (Archit Singla) wrote…</i></summary><blockquote>

can you check what if this value is made to 0, will it handle? if not handle it
</blockquote></details>

This is handled by TimeUnit class itself, doesn't need to be handled explicitly
Verified by running locally as well.

**Comment by shalini-agr** [comment]
trigger uts

**Comment by shalini-agr** [comment]
trigger copyrightcheck

**Comment by shalini-agr** [comment]
trigger uts

**Comment by shalini-agr** [comment]
trigger ut2

**Review by archit-harness — APPROVED** [review]
> Reviewed 1 of 12 files at r5, 1 of 1 files at r6, 1 of 1 files at r7, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/50353)* status: 21 file(s) to review. 1. unresolved discussions. 1 of 2 approvals obtained.

**Review by ankittiwari-harness — APPROVED** [review]

File: 120-ng-manager/config.yml [hunk 3]
```diff
@@ -541,4 +541,12 @@ queueServiceClientConfig:
   envNamespace: ${EVENTS_FRAMEWORK_ENV_NAMESPACE:-localhost}
   topic: ${HSQS_TOPIC:-ng}
 
+webhookPushEventHsqsDequeueConfig:
```
Comment (sribalij) [inline 120-ng-manager/config.yml:544]: Are these numbers common at framework level?

**Review by shalini-agr — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/50353)* status: 21 file(s) to review. 2. unresolved discussions. 

___
*[`120-ng-manager/config.yml` line 544 at r7](https://reviewable.io/reviews/harness/harness-core/50353#gh-1266657565:-N_cr9mjAkzq_9BgLrRO:b-1sa3on) ([raw file](https://github.com/harness/harness-core/blob/a1e2e9da38976a07fd006c6390bb9d6bbf3bda7e/120-ng-manager/config.yml#L544)):*
<details><summary><i>Previously, sribalij (Sridhar) wrote…</i></summary><blockquote>

Are these numbers common at framework level?
</blockquote></details>

No, these are not common at the framework level

we have to configure these numbers based on our use case. We can tune them based on our tests.

**Comment by shalini-agr** [comment]
trigger review

**Review by sribalij — APPROVED** [review]

**Pull request merged**